### PR TITLE
Switch to using aws-java-sdk-bundle dependency

### PR DESCRIPTION
**Pull request by samarthjain**
AWS provides bundled java-sdk dependency (https://aws.amazon.com/blogs/developer/java-sdk-bundle/) which shades jackson and other dependencies that was preventing druid from upgrading from 1.10.* version of aws-sdk. This pull request causes druid to now use this bundle as a dependency.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1171,7 +1129,7 @@
             <id>spark2</id>
             <properties>
                 <jackson.version>2.6.5</jackson.version>
-                <aws.sdk.version>1.11.143</aws.sdk.version>
+                <aws.java.sdk.bundle.version>1.11.143</aws.java.sdk.bundle.version>
```
Comment (leventov) [inline pom.xml]: This line not needed

File: pom.xml [hunk 2]
```diff
@@ -78,9 +78,8 @@
         <hadoop.compile.version>2.7.3</hadoop.compile.version>
         <hive.version>2.0.0</hive.version>
         <powermock.version>1.6.6</powermock.version>
-        <!-- Cannot update to AWS SDK 1.11+ because of Jackson incompatibility.
-        Need to update Druid to use Jackson 2.6+ -->
-        <aws.sdk.version>1.10.77</aws.sdk.version>
+        <!-- Use aws-sdk bundle to let druid use newer version of aws-sdk and handle dependency conflict with Jackson -->
+        <aws.java.sdk.bundle.version>1.11.143</aws.java.sdk.bundle.version>
```
Comment (leventov) [inline pom.xml]: Not needed to extract this as a property, because it's used just at one place

**Comment by leventov** [comment]
@samarthjain thanks for contribution, please sign CLA: http://druid.io/community/cla.html

**Comment by samarthjain** [comment]
@jihoonson - I am closing this pull request for now.

**Comment by jihoonson** [comment]
Ok, thanks.

**Pull request closed without merging**